core/station/impl/src/services/system.rs

```diff
@@ -575,69 +662,257 @@ mod init_canister_sync_handlers {
+                .await?;
+        }
+
+        Ok(())
+    }
+
+    pub fn set_initial_named_rules(named_rules: &[InitNamedRuleInput]) -> Result<(), ApiError> {
+        let mut add_named_rules = named_rules
+            .iter()
+            .map(|named_rule| {
+                let input = AddNamedRuleOperationInput {
+                    name: named_rule.name.clone(),
+                    description: named_rule.description.clone(),
+                    rule: named_rule.rule.clone().into(),
+                };
+
+                (
+                    input,
+                    HelperMapper::to_uuid(named_rule.id.clone()).map(|uuid| *uuid.as_bytes()),
+                )
+            })
+            .map(|(input, result)| result.map(|uuid| (input, uuid)))
+            .collect::<Result<Vec<_>, _>>()?;
+
+        // sorting criteria:
+        // - if a policy depends on another policy, the dependent policy should be added first
+        // - keep the original order of the policys otherwise
+        add_named_rules.sort_by(|a, b| {
+            if b.0.rule.has_named_rule_id(&a.1) {
+                Ordering::Less
+            } else {
+                Ordering::Greater
+            }
+        });
+
+        for (new_named_rule, with_named_rule_id) in add_named_rules {
+            NAMED_RULE_SERVICE.create_with_id(new_named_rule, Some(with_named_rule_id))?;
+        }
+
+        Ok(())
+    }
+
+    pub async fn set_initial_permissions(
+        permissions: &[InitPermissionInput],
+    ) -> Result<(), ApiError> {
+        for permission in permissions {
+            let users = permission
+                .allow
+                .users
+                .iter()
+                .map(|id| HelperMapper::to_uuid(id.clone()).map(|uuid| *uuid.as_bytes()))
+                .collect::<Result<Vec<_>, _>>()?;
+
+            let user_groups = permission
+                .allow
+                .user_groups
+                .iter()
+                .map(|id| HelperMapper::to_uuid(id.clone()).map(|uuid| *uuid.as_bytes()))
+                .collect::<Result<Vec<_>, _>>()?;
+
+            let input = EditPermissionOperationInput {
+                resource: permission.resource.clone().into(),
+                auth_scope: Some(permission.allow.auth_scope.clone().into()),
+                users: Some(users),
+                user_groups: Some(user_groups),
+            };
+
+            PERMISSION_SERVICE.edit_permission(input)?;
+        }
+
+        Ok(())
+    }
+
+    fn specifier_has_reference_to_policy_id(
+        specifier: &RequestSpecifier,
+        policy_id: &UUID,
+    ) -> bool {
+        match specifier {
+            RequestSpecifier::EditRequestPolicy(resource_ids)
+            | RequestSpecifier::RemoveRequestPolicy(resource_ids) => match resource_ids {
+                ResourceIds::Any => false,
+                ResourceIds::Ids(ids) => ids.contains(policy_id),
+            },
+            _ => false,
```

Comment:
Let's please unfold the wildcard so that the compiler forces us to update this definition upon changes to `RequestSpecifier` type.

Reply:
replacing Greater with Equal, since the sort is stable, it should keep the original order for independent named rules/policies

Reply:
Good idea!